posts/web-scraping-using-lxml-python.md

```diff
@@ -0,0 +1,208 @@
+Hello everyone! I hope you are doing well. I am [Yasoob](http://yasoob.me) from [Python Tips](https://pythontips.com). In this article, I'll teach you the basics of web scraping using Python. First of all, why should you even bother learning how to web scrape? If your job doesn't require you to learn it, then let me give you some motivation. What if you want to create a website which curates cheapest products from Amazon, Walmart and a couple of other online stores? A lot of these online stores don't provide you with an easy way to access their information using an API. In the absence of an API, your only choice is to create a web scraper which can extract information from these websites automatically and provide you with that information in an easy to use way.
+
+Here is an example of a typical API response in JSON. This is the response from Reddit:
+
+![Typical API Response in JSON](https://imgur.com/WOBUbyn.png)
+
+There are a lot of Python libraries out there which can help you with web scraping. There is [lxml](http://lxml.de/), [BeautifulSoup](https://www.crummy.com/software/BeautifulSoup/?) and a full-fledged framework called [Scrapy](https://scrapy.org/). Most of the tutorials discuss lxml and Scrapy, so I decided to go with lxml in this post. I will teach you the basics of XPaths and how you can use them to extract data from an HTML document. I will take you through a couple of different examples so that you can quickly get up-to-speed with lxml and XPaths.
```

Comment:
I believe you said 'Most of the tutorials discuss BeautifulSoup and Scrapy'

Reply:
I guess I must have been dozing while writing it. Thanks! I will fix it